### ISSUE TITLE
rclone lsd -P deadlocks

### ISSUE THREAD

**Issue by nicren** [rclone/rclone#7102]
#### The associated forum post URL from `https://forum.rclone.org`

n/a

#### What is the problem you are having with rclone?

Commands `ls`, `lsd`, `lsd` do not display anything & hang up when called under PowerShell v7.3.5.
Nonetheless, all 3 commands work OK when called from Windows 10 regular command prompt.

NB :
- command `lsjson` works OK under PowerShell v7.3.5 & the regular command prompt;
- identical behaviour when using PowerShell v7.3.5 directly or within Windows Terminal v1.17.11461.0;
- other commands do not appear affected eg. `sync` & `serve` work OK under PowerShell v7.3.5

#### What is your rclone version (output from `rclone version`)

v1.63.0

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Windows 10, 64-bit

#### Which cloud storage system are you using? (e.g. Google Drive)

Same behaviour with local, OneDrive, Swift storages.

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`rclone ls local:d `, `rclone lsd local:d`, `rclone lsf local:d`, `rclone ls remote:d `, `rclone lsd remote:d`, `rclone lsf remote:d`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2023/07/01 11:46:01 DEBUG : Setting --stats "0s" from environment variable RCLONE_STATS="0"
2023/07/01 11:46:01 DEBUG : Setting --verbose "1" from environment variable RCLONE_VERBOSE="1"
2023/07/01 11:46:01 DEBUG : Setting --stats-one-line "true" from environment variable RCLONE_STATS_ONE_LINE="true"
2023/07/01 11:46:01 DEBUG : Setting --progress "true" from environment variable RCLONE_PROGRESS="true"
2023/07/01 11:46:01 DEBUG : Setting --filter-from "[D:\\filter.rclone]" from environment variable RCLONE_FILTER_FROM="D:\\filter.rclone"
2023/07/01 11:46:02 DEBUG : rclone: Version "v1.63.0" starting with parameters ["C:\\Program Files\\rclone\\rclone.exe" "ls" "remote:tmp" "-vv"]
2023/07/01 11:46:02 DEBUG : Creating backend with remote "remote:tmp"
2023/07/01 11:46:02 DEBUG : Using config file from "C:\\Users\\\\.config\\rclone\\rclone.conf"
2023/07/01 11:46:02 DEBUG : Creating backend with remote "remote:backups/tmp"
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
I managed to replicate this.

It is to do with the `--progress` flag you are setting in the environment.

For some reason this locks up just as you say under powershell.

    ./rclone.exe lsd -P .

It also locks up under cmd and on linux so I suspect you didn't have the `RCLONE_PROGRESS` env var set in your cmd test.

Try this and see if it fixes it

[v1.64.0-beta.7121.d73e0f1de.fix-7102-stdout-deadlock](https://beta.rclone.org/branch/fix-7102-stdout-deadlock/v1.64.0-beta.7121.d73e0f1de.fix-7102-stdout-deadlock/) on branch [fix-7102-stdout-deadlock](https://github.com/rclone/rclone/tree/fix-7102-stdout-deadlock) (uploaded in 15-30 mins)

**Comment by nicren** [comment]
Hi again,

Thanks a lot for your quick answer (&, by the way, for your extraordinary software which is useful to me every single day!).

I confirm that I had set the `RCLONE_PROGRESS` env. var. to `true`.
After removing it, all commands & storages mentioned in my initial report work OK in rclone v1.63.0 in PowerShell. Conversly, all break down again if I set the `-P` flag.

On the other hand, all mentioned commands & storages work OK under the rclone v1.64.0beta1 you provided, whether setting the env. var. or flag.

Thanks again for your help.
